File: JbmrksIOs/JbmrksIOs/Models/DashboardStats.swift
//
//  DashboardStats.swift
//  JbmrksIOs
//
//  Dashboard statistics model
//

import Foundation

struct DashboardStats {
    let activeTasks: Int
    let completedToday: Int
    let unreadMessages: Int
    let upcomingEvents: Int
    let userName: String
    let recentActiveTasks: [Task]
    let recentCompletedTasks: [Task]
    
    init(
        activeTasks: Int = 0,
        completedToday: Int = 0,
        unreadMessages: Int = 0,
        upcomingEvents: Int = 0,
        userName: String = "User",
        recentActiveTasks: [Task] = [],
        recentCompletedTasks: [Task] = []
    ) {
        self.activeTasks = activeTasks
        self.completedToday = completedToday
        self.unreadMessages = unreadMessages
        self.upcomingEvents = upcomingEvents
        self.userName = userName
        self.recentActiveTasks = recentActiveTasks
        self.recentCompletedTasks = recentCompletedTasks
    }
}

File: app/src/main/java/com/example/jbmarks/dashboard/data/DashboardStats.kt
package com.example.jbmarks.dashboard.data

import com.example.jbmarks.tasks.domain.Task

/**
 * Dashboard statistics data
 */
data class DashboardStats(
    val activeTasks: Int = 0,
    val completedToday: Int = 0,
    val unreadMessages: Int = 0,
    val upcomingEvents: Int = 0,
    val userName: String = "User",
    val recentActiveTasks: List<Task> = emptyList(),
    val recentCompletedTasks: List<Task> = emptyList()
)
